src/Application.vala

```diff
@@ -190,70 +186,58 @@ namespace Switchboard {
             main_window.bind_property ("title", title_label, "label");
 
             shutdown.connect (() => {
-                if (plug_widgets[leaflet.visible_child] != null && plug_widgets[leaflet.visible_child] is Switchboard.Plug) {
-                    plug_widgets[leaflet.visible_child].hidden ();
+                var plug = plug_widgets[navigation_view.visible_page.child];
+                if (plug != null && plug is Switchboard.Plug) {
+                    plug.hidden ();
                 }
             });
 
-            leaflet.notify["visible-child"].connect (() => {
+            navigation_view.popped.connect (() => {
                 update_navigation ();
             });
 
-            leaflet.notify["child-transition-running"].connect (() => {
+            navigation_view.pushed.connect (() => {
                 update_navigation ();
             });
         }
 
         private void update_navigation () {
-            if (!leaflet.child_transition_running) {
-                if (plug_widgets[leaflet.get_adjacent_child (Adw.NavigationDirection.FORWARD)] != null) {
-                    plug_widgets[leaflet.get_adjacent_child (Adw.NavigationDirection.FORWARD)].hidden ();
-                }
-
-                var previous_child = plug_widgets[leaflet.get_adjacent_child (Adw.NavigationDirection.BACK)];
-                if (previous_child != null && previous_child is Switchboard.Plug) {
-                    previous_child.hidden ();
-                }
+            var next_plug = plug_widgets[navigation_view.get_next_page ().child];
```

Comment:
This line generates the warning `Adwaita-CRITICAL **: 18:07:02.523: adw_navigation_page_get_child: assertion 'ADW_IS_NAVIGATION_PAGE (self)' failed` when navigating away from the home page, presumably because `get_next_page ()` returns `null`.  Navigation does occur as expected though.